src/component/common/Dialogue/Dialogue.tsx

```diff
@@ -61,12 +69,14 @@ const Dialogue: React.FC<IDialogue> = ({
                     condition={Boolean(onClick)}
                     show={
                         <Button
+                            form={formId}
                             color="primary"
                             variant="contained"
-                            onClick={onClick}
-                            autoFocus
+                            onClick={handleClick}
+                            autoFocus={!formId}
                             disabled={disabledPrimaryButton}
                             data-test={DIALOGUE_CONFIRM_ID}
+                            type={formId ? 'submit' : undefined}
```

Comment:
Why undefined here instead of button? I think material ui will resolve button as type if it is undefined, but if it does we might as well skip the implicit overhead and be explicit?

Reply:
That's a good point. I did initially set it to be 'button' instead of undefined because that's what Material does by default. However, I decided to change it to `undefined` in the event that Material changes its defaults later. Maybe they'd like to use a different type instead. If we pass `undefined`, then we let Material choose for us.

However, I also don't see this happening any time soon (why would it?), so I'm happy to change it back to 'button' if you think that's better 💁🏾 . 

One final point, though: I reckon most uses of Material buttons won't define the type at all, so it'll be 'undefined' in a large majority of cases. They have probably optimized well for that use case, so I don't expect it to have much of an impact (but I haven't measured it!).

Reply:
I'll merge this now, so I just made the change. Those are fair points, I think it is entirely fine to pass undefined and let material set the button type in most use cases, however when we elect to explicitly set the type, it seems counter-intuitive to me to pass undefined as a value when the spec is explicit in terms of accepted values. 

In the event that material changes how they handle internal type settings, I don't think they will ever elect to override a specific choice that complies with the HTML spec.

When it comes to performance, it depends on how they have implemented it. But it's still such a small optimization that I don't think it's even worth measuring.

Reply:
Sure; I can get on board with that! 😄 Just merge at your convenience, then 🚂